internal/server/event_server.go

```diff
@@ -129,7 +129,9 @@ func (s *EventServer) logRateLimitMiddleware(h http.Handler) http.Handler {
 			s.logger.V(1).Info("Discarding event, rate limiting duplicate events",
 				"reconciler kind", event.InvolvedObject.Kind,
 				"name", event.InvolvedObject.Name,
-				"namespace", event.InvolvedObject.Namespace)
+				"namespace", event.InvolvedObject.Namespace,
+				"event metadata", event.Metadata,
+				"rate limiting key", getEventKey(event))
```

Comment:
> Regarding the metadata, since only the revision or only a few specific keys are considered for rate limiting, does printing the whole content of the metadata map help with anything for such discarded event log?

It's how I found out the bug. I printed the un-hashed event key and the event metadata. I couldn't see the revision anywhere in the event key, but I could see the revision in the metadata with the key `.../revision`. I didn't know that helm-controller was prefixing the `revision` key with something, so when I saw the event metadata I could immediately spot why the revision was missing in the event key. That's when I realized that there was probably some pre-processing to cleanup the metadata, because earlier I saw the `revision` key on the Helm Promotion guide in the GitHub Repository Workflow context object (i.e. `${{ github.event.client_payload.metadata.revision }}`). If I had just printed `event.Metadata[eventv1.MetaRevisionKey]` I wouldn't have seen anything useful, just a behavior consistent with the revision missing in the event key.

This is a debug-level log, and for me the ideal debug-level log would actually print the entire event payload, not just the metadata. But I'm more than happy to remove this, what I would really love to see fixed ASAP is just the bug itself :ok_hand:

Reply:
> only in debug mode

isn't that already the current situation for the log I touched? due to `.V(1).Info` :thinking:

Reply:
I'm gonna leave this debug log improvement to another PR :ok_hand: